docs/appendix.rst

```diff
@@ -18,112 +18,77 @@ This documentation uses a few special terms to refer to Python types:
-      CIM Instance, etc.  It is one of the output formats provided for
+      CIM instances, etc.  It is one of the output formats provided for
       the display of CIM objects in pywbemcli. See DMTF term:`DSP0004` for more
       information on the MOF format.
 
    WBEM management profile
       The DMTF and SNIA define specific profiles of manageability that are
       published as specifications in the DMTF and within the SMI-S specification
       in SNIA. A management profile defines the manageability characteristics
-      of a specic set of services in terms of the CIM model and WBEM operations.
+      of a specific set of services in terms of the CIM model and WBEM operations.
       These profiles are documented by name and version and are incorporated into
       compliant WBEM servers so that they can be discovered by WBEM clients to
       determine the management capabilities of the WBEM server.
       See :ref:`Profile advertisement methodologies`
 
-   WBEM-URI
-      The wbem-uri is a standardized text form for CIM instance names. It is
-      documented in DMTF :term:`DSP0207`. Pywbemcli uses the untyped WBEM URI
-      as the format for instance names in CLI input parameters
-      (i.e. :term:`INSTANCENAME`)::
-
-            WBEM-URI = WBEM-URI-UntypedNamespacePath /
-                       WBEM-URI-UntypedClassPath /
-                       WBEM-URI-UntypedInstancePath
-
-            WBEM-URI-UntypedNamespacePath = namespacePath
-            WBEM-URI-UntypedClassPath     = namespacePath ":" className
-            WBEM-URI-UntypedInstancePath  = WBEM-URI-UntypedInstancePath
-                                            className "." key_value_pairs
-
-            namespacePath = [namespaceType ":"] namespaceHandle
-            namespaceType = ("http" ["s"]) / ("cimxml.wbem" ["s"])
-            namespaceHandle = ["//" authority] "/" [namespaceName]
-            namespaName     = IDENTIFIER *("/"IDENTIFIER))
-
-            // Untyped key value pairs
-            key_value_pairs  = key_value_pair *("," key_value_pair)
-            key_value_pair   = key_name "=" key_value
-            key_value        = stringValue / charValue / booleanValue /
-                               integerValue / realValue /
-                               "\"" datetimeValue "\"" /
-                               "\"" referenceValue "\""
-
-      In pywbemcli the WBEM-URI is used as the format for instance names on
-      commands such as ``instance get <instance-name>``
-
-      In these cases, the normal use is to specify only the classname and
-      keybindings so that examples of valid WBEM-URIs would be::
-
-        CIM_RegisteredProfile.InstanceID="acme:1"
-        CIM_RegisteredProfile.InstanceID=100
+   WBEM URI
+      WBEM URI is a standardized text form for CIM object paths and is
+      defined in :term:`DSP0207`. Pywbemcli uses the untyped WBEM URI format
+      for instance names in the command line (i.e. :term:`INSTANCENAME`).
```

Comment:
The ABNF that you had there was already too complex. Plus, at a closer look it had at least one error (definition of WBEM-URI-UntypedInstancePath uses itself).

If we have any ABNF anywhere, i suggest that we reduce it to just what is needed for INSTANCENAME. Because that is the only place we need it in pywbemcli. Then we can restrict it to instances only, to untyped only, and can omit the host part.

That leaves something like this:

```
INSTANCENAME = CLASSNAME "." keybindings
keybindings = keybinding *( "," keybinding )
keybinding = PROPERTYNAME "=" value
value = constantValue / "\"" escaped_instancename "\""
```
where `constantValue` is defined in DSP0004, and `escaped_instancename` is a backslash-escaped `INSTANCENAME`.

Reply:
I agree.  Some wordsmithing needed.

We propose to add this to the section on on how interactive instance name selection

Reply:
The new section will be added as part of issue #282.